Core/include/Acts/Geometry/CuboidPortalShell.hpp

```diff
@@ -0,0 +1,175 @@
+  std::string label() const override;
+
+  const Transform3& transform() const override {
+    return m_volume->transform();
+  };
+
+ private:
+  std::array<std::shared_ptr<Portal>, 6> m_portals{};
+
+  TrackingVolume* m_volume;
+};
+
+/// @class CuboidStackPortalShell
+/// This class describes a cuboid shell containing multiple volumes.
+class CuboidStackPortalShell : public CuboidPortalShell {
+ public:
+  /// Construct the portal shell stack from the given shells
+  /// @param gctx The geometry context
+  /// @param shells The shells to stack
+  /// @note The shells must be ordered in the given direction
+  /// @param axis The stacking direction (along x/y/z axis) in local stack coordinates
+  /// @param logger A logging instance for debugging
+  CuboidStackPortalShell(const GeometryContext& gctx,
+                         std::vector<CuboidPortalShell*> shells,
+                         AxisDirection axis,
+                         const Logger& logger = getDummyLogger());
+
+  /// Construct the portal shell stack from the given shells
+  /// @param gctx The geometry context
+  /// @param shells The shells to stack
+  /// @note The shells must be ordered in the given direction
+  /// @param direction The stacking direction in global coordinates
+  /// @param logger A logging instance for debugging
+  CuboidStackPortalShell(const GeometryContext& gctx,
+                         std::vector<CuboidPortalShell*> shells,
+                         const Vector3& direction,
+                         const Logger& logger = getDummyLogger());
+
+  /// @copydoc PortalShellBase::size
+  std::size_t size() const final;
+
+  /// @copydoc CuboidPortalShell::portal
+  Portal* portal(Face face) final;
+
+  /// @copydoc CuboidPortalShell::portalPtr
+  std::shared_ptr<Portal> portalPtr(Face face) final;
+
+  /// @copydoc CuboidPortalShell::setPortal
+  void setPortal(std::shared_ptr<Portal> portal, Face face) final;
+
+  void applyToVolume() override {
+    // No-op, because it's a composite portal shell
+  }
+
+  /// @copydoc PortalShellBase::isValid
+  bool isValid() const final;
+
+  /// @copydoc PortalShellBase::label
+  std::string label() const final;
+
+  /// Return the stack's group transform
+  const Transform3& transform() const override;
+
+  /// Convert a global vector to an axis direction in local stack coordinates
+  /// @param dir is the global direction to convert
+  static AxisDirection directionToAxis(const Vector3& dir);
+
+ private:
+  void stackShell(const GeometryContext& gctx, const Logger& logger);
+
+  /// Shell stacking direction in global coordinates
+  Vector3 m_direction;
+
+  /// Shell stacking direction in local stack coordinates
+  AxisDirection m_axis;
+
+  /// The cuboid face positioned first along the stacking direction
+  CuboidVolumeBounds::Face m_frontFace = negativeYZPlane;
+  /// The cuboid face positioned last along the stacking direction
+  CuboidVolumeBounds::Face m_backFace = positiveYZPlane;
```

Comment:
The right hand side is `BoundarySurfaceFace`, right? Maybe we could initialize them directyly from `Face`?